src/log.h

```diff
@@ -221,16 +279,62 @@ class StreamProxy {
 public:
 	StreamProxy(std::ostream *os) : m_os(os) { }
 
+	inline static void maybe_fix_stream_state(std::ostream &os)
```

Comment:
By `inline`, this function would be embedded into every type of `StreamProxy& operator<<(T&& arg)`, which potentially (for GCC it does) produce a lot of duplicated code, resulting in possibly longer compile times and larger binary.

Reply:
The alternative (doing an actual function call to a non-inlineable function) is worse IMO.

Reply:
The benefit of inlining is that it can be optimized out when the const char* is known to be non-null by the compiler.

Reply:
I will move the `os.good()` check outside so that it is always inlined, and put the rest of the code in a non-inlined method. That avoids the downside of inlining the entire thing.